cclient/iota_client_extended_api.c

```diff
@@ -56,6 +56,107 @@ static retcode_t is_unused_address(iota_client_service_t const* const serv,
   return ret_code;
 }
 
+static retcode_t traverse_bundle(iota_client_service_t const* const serv,
+                                 flex_trit_t const* const tail_hash,
+                                 bundle_transactions_t* const bundle,
+                                 hash8019_array_p trytes) {
+  retcode_t ret_code = RC_OK;
+  get_trytes_req_t* get_trytes_req = NULL;
+  get_trytes_res_t* get_trytes_res = NULL;
+  iota_transaction_t tx = {};
+  flex_trit_t* tmp_trytes = NULL;
+  flex_trit_t trunk_hash[FLEX_TRIT_SIZE_243];
+  int64_t current_index = 0, last_index = 0;
+  bool is_tail = true;
+
+  log_info(logger_id, "[%s:%d]\n", __func__, __LINE__);
+  get_trytes_req = get_trytes_req_new();
+  if (!get_trytes_req) {
+    ret_code = RC_CCLIENT_OOM;
+    log_error(logger_id, "%s get_trytes_req_new failed: %s\n", __func__,
+              error_2_string(ret_code));
+    goto cleanup;
+  }
+  get_trytes_res = get_trytes_res_new();
+  if (!get_trytes_res) {
+    ret_code = RC_CCLIENT_OOM;
+    log_error(logger_id, "%s get_trytes_res_new failed: %s\n", __func__,
+              error_2_string(ret_code));
+    goto cleanup;
+  }
+
+  do {
+    memcpy(trunk_hash, is_tail ? tail_hash : transaction_trunk(&tx),
```

Comment:
we don't need the local `trunk_hash` here, can simply call `transaction_set_trunk(tx, tail_hash)` before the loop and always refer to the hash inside the loop via `transaction_trunk`

Reply:
`trunk_hash` log the bundle hash for next transaction that we don't recalculate it during deserialization.

Reply:
no, it holds the tail tx's hash (i.e. H(tx in bundle with curIdx=0), ...)

Reply:
done.